web/src/queries/software.ts

```diff
@@ -367,12 +401,34 @@ const useProposalChanges = () => {
   }, [client, queryClient]);
 };
 
+/**
+ * Hook that registers a useEffect to listen for conflicts changes
+ *
+ */
+const useConflictsChanges = () => {
+  const client = useInstallerClient();
+  const queryClient = useQueryClient();
+  React.useEffect(() => {
+    if (!client) return;
+
+    return client.onEvent((event) => {
+      if (event.type === "ConflictsChanged") {
+        const { conflicts } = event;
+        queryClient.setQueryData([conflictsQuery().queryKey], conflicts);
```

Comment:
I'm not sure if it is a good idea to send the complete list of the conflicts as a part of the event via the web socket. There might be dozens of conflicts and sending them to all clients sounds too much to me. Not all clients might be interested in the details, like the `agama monitor` client.

I'd prefer no details in the event and let the client to fetch the data again if it needs them.  With Tanstack query it is enough to invalidate the cache, it will re-fetch automatically.

Something like:

```js
        queryClient.invalidateQueries({ queryKey: [conflictsQuery().queryKey] })
```

See https://tanstack.com/query/v5/docs/framework/react/guides/query-invalidation

Reply:
I guess that apart of changing the frontend for just invalidating the query some changes are needed in the backend for not broadcasting all the conflicts

Reply:
Yes, the backend needs an update as well.